packages/headless/src/invoices.ts

```diff
@@ -15,6 +15,27 @@ import { MutationKey, QueryKey } from "@/constants";
 type OnSuccess = Void;
 type OnError = (error: Error) => void;
 
+type useFindInvoicesProps = {
+  query: UseInfiniteQueryResult<
+    InfiniteData<Paginated<IInvoice.Self>, unknown>,
+    Error
+  >;
+  list: IInvoice.Self[] | null;
+  more: () => void;
+};
+
+export function useFindInvoices({
+  pageParam,
+}: {
+  pageParam: number;
+}): useFindInvoicesProps {
+  const atlas = useAtlas();
+  const findInvoices = useCallback(async () => {
+    return await atlas.invoice.find({ page: pageParam });
+  }, []);
+  return usePaginationQuery(findInvoices, [QueryKey.FindInvoices]);
```

Comment:
Use the `usePaginate` hook isntead of `usePaginationQuery`. 

```suggestion
  return usePaginate(findInvoices, [QueryKey.FindInvoices]);
```

Here is an example: 

https://github.com/litespace-org/litespace/blob/0795a725442639fd1ef03bb212a4304766f8577b/packages/headless/src/users.ts#L7-L16